spec/class.dd

```diff
@@ -430,25 +430,29 @@ $(H2 $(LNAME2 class-instantiation, Class Instantiation))
 
 $(H3 $(LNAME2 constructors, Constructors))
 
+        $(P A constructor is a special method which is normally
+        $(RELATIVE_LINK2 instantiation-process, invoked by the compiler)
+        when the class instance is created.)
+
 $(GRAMMAR
 $(GNAME Constructor):
     $(D this) $(GLINK2 function, Parameters) $(GLINK2 function, MemberFunctionAttributes)$(OPT) $(GLINK2 function, FunctionBody)
     $(D this) $(GLINK2 function, Parameters) $(GLINK2 function, MemberFunctionAttributes)$(OPT) $(GLINK2 function, MissingFunctionBody)
     $(GLINK2 template, ConstructorTemplate)
 )
 
-        $(P Constructors are defined with a function name of $(D this)
-        and have no return value:)
+        $(P A constructor is declared with a function name of $(D this)
+        and no return type:)
 
         $(SPEC_RUNNABLE_EXAMPLE_RUN
         ------
         class A
         {
             int i;
 
-            this(int i) // constructor taking an int argument
+            this(int val) // constructor taking an int argument
             {
-                this.i = i; // initialize field `i` from parameter `i`
```

Comment:
Is there a still example of `this.i = i` somewhere else? That's a common pattern also worth noting as officially supported.

Reply:
I don't think so, not nearby anyway. Updated so that example is kept and added a constructor overload to demonstrate member variables being in scope.